lib/freeze.js

```diff
@@ -37,42 +34,53 @@ module.exports = class Freeze {
     });
   }
 
-  freeze(context, props) {
+  async freeze(context, props) {
     const labels = context.payload.issue.labels;
     if (GitHubHelper.isLabelOnIssue(
       context.payload.issue,
       this.config.labelName) === undefined) {
       labels.push(this.config.labelName);
     }
 
-    this.github.issues.edit(context.issue({
+    const kv = metadata(context.github, context.issue(), process.env.APP_ID);
+    await kv.set('snooze', props);
+
+    await this.github.issues.edit(context.issue({
       state: 'closed',
       labels
     }));
 
-    this.github.issues.createComment(context.issue({
+    await this.github.issues.createComment(context.issue({
       body: 'Sure thing. I\'ll close this issue for a bit. I\'ll ping you around ' +
-      moment(props.unfreezeMoment).calendar() + ' :clock1: ' +
-      '<!-- ' + JSON.stringify(props) + '-->'
+      moment(props.unfreezeMoment).calendar() + ' :clock1: '
     }));
   }
 
-  unfreeze(issue, props) {
+  async checkUnfreeze(issue) {
+    const kv = metadata(this.github, issue, process.env.APP_ID);
+    const props = await kv.get('snooze');
+
+    if (moment(props.unfreezeMoment) < moment()) {
+      return this.unfreeze(issue, props);
```

Comment:
`checkUnfreeze` to me (YMMV, interested in your thoughts) implies a method that returns a result of the check. Does unfreezing here make this mutating? should this just return true/false, and we let the calling function call unfreeze?

Reply:
Yeah, the reason I moved this all inline was so the prop parsing could all be hidden. Otherwise, the caller of `freeze` needs to have access to the props. I can experiment with cleaning this up.

Reply:
That makes sense, and is cleaner to not deal with passing props around. Maybe `rename` it then?  Check seems to he the wrong verb now. Just `unfreeze` sounds overkill, but I think it's better.